frontend/src/metabase/common/hooks/use-has-models.tsx

```diff
@@ -0,0 +1,17 @@
+import type { SearchRequest } from "metabase-types/api";
+
+import { useFetchModels } from "./use-fetch-models";
+
+/** NOTE: hasModels is undefined when the request is pending */
+export const useHasModels = (req: Partial<SearchRequest> = {}) => {
+  const modelsResult = useFetchModels({
+    limit: 1,
+    model_ancestors: false,
+    ...req,
+  });
+  const modelsLength = modelsResult.data?.data.length;
+  return {
+    hasModels: modelsLength === undefined ? undefined : modelsLength !== 0,
+    modelsResult,
```

Comment:
I like having a dedicated hook for this, but given that it has a very narrow purpose, why are we passing the whole data object back? I would expect this to either (a) just return a boolean or (b) a boolean and an `isLoading` prop. Inferring isLoading from undefined state is a lot harder to read and follow, and makes this code less easily reusable